core/os/os.h

```diff
@@ -77,6 +77,8 @@ class OS {
 	void _set_logger(CompositeLogger *p_logger);
 
 public:
+	const String UNNAMED_PROJECT_NAME = "[unnamed godot project]";
```

Comment:
That feels a bit overkill to have something not really OS-related as a constant here. Since it's only used in two locations, I'd suggest just hardcoding the value in each class.

Reply:
I'd also suggest simplifying to just "[unnamed project]".